code/game/gamemodes/changeling/powers/augmented_eyesight.dm

```diff
@@ -3,84 +3,54 @@
 
 /datum/action/changeling/augmented_eyesight
 	name = "Augmented Eyesight"
-	desc = "Creates heat receptors in our eyes and dramatically increases light sensing ability."
-	helptext = "Grants us thermal vision or flash protection. We will become a lot more vulnerable to flash based devices while thermal vision is active."
+	desc = "Creates more light sensing rods in our eyes, allowing our vision to penetrate most blocking objects. Protects our vision from flashes while inactive."
+	helptext = "Grants us x-ray vision or flash protection. We will become a lot more vulnerable to flash-based devices while x-ray vision is active."
 	button_icon_state = "augmented_eyesight"
 	chemical_cost = 0
 	dna_cost = 2 //Would be 1 without thermal vision
 	active = FALSE
 
 /datum/action/changeling/augmented_eyesight/on_purchase(mob/user) //The ability starts inactive, so we should be protected from flashes.
+	if(!istype(user))
+		return
 	..()
-	var/obj/item/organ/internal/cyberimp/eyes/O = new /obj/item/organ/internal/cyberimp/eyes/shield/ling()
-	O.insert(user)
-	active = FALSE
+	var/obj/item/organ/internal/eyes/E = user.get_organ_slot("eyes")
+	if (E)
+		E.flash_protect = FLASH_PROTECTION_WELDER //Adjust the user's eyes' flash protection
+		to_chat(user, "We adjust our eyes to protect them from bright lights.")
+	else
+		to_chat(user, "We can't adjust our eyes if we don't have any!")
 
 /datum/action/changeling/augmented_eyesight/sting_action(mob/living/carbon/user)
 	if(!istype(user))
 		return
-	if(!active)
-		var/obj/item/organ/internal/cyberimp/eyes/O = new /obj/item/organ/internal/cyberimp/eyes/thermals/ling()
-		O.insert(user)
-		active = TRUE
-	else
-		var/obj/item/organ/internal/cyberimp/eyes/O = new /obj/item/organ/internal/cyberimp/eyes/shield/ling()
-		O.insert(user)
-		active = FALSE
-	return 1
-
-/datum/action/changeling/augmented_eyesight/Remove(mob/user)
-	var/obj/item/organ/internal/cyberimp/eyes/O = user.get_organ_slot("eye_ling")
-	if(O)
-		O.remove(user)
-		qdel(O)
-	..()
-
-/obj/item/organ/internal/cyberimp/eyes/shield/ling
-	name = "protective membranes"
-	desc = "These variable transparency organic membranes will protect you from welders and flashes and heal your eye damage."
-	icon_state = "ling_eyeshield"
-	eye_colour = "#000000"
-	implant_overlay = null
-	slot = "eye_ling"
-	status = 0
-	aug_message = "We feel a minute twitch in our eyes, our eyes feel more durable."
-
-/obj/item/organ/internal/cyberimp/eyes/shield/ling/on_life()
 	..()
-	var/obj/item/organ/internal/eyes/E = owner.get_int_organ(/obj/item/organ/internal/eyes)
-	if(owner.eye_blind || owner.eye_blurry || (BLINDNESS in owner.mutations) || (NEARSIGHTED in owner.mutations) || (E && E.damage > 0))
-		owner.reagents.add_reagent("oculine", 1)
```

Comment:
Undocumented change that changeling thermal eyes will no longer cure eye damage.